extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityProcessor.java

```diff
@@ -85,6 +112,158 @@ void registerSecurityInterceptors(BuildProducer<InterceptorBindingRegistrarBuild
         beans.produce(new AdditionalBeanBuildItem(SecurityHandler.class, SecurityConstrainer.class));
     }
 
+    @BuildStep
+    void gatherSecurityChecks(BuildProducer<BeanRegistrarBuildItem> beanRegistrars,
+            ApplicationIndexBuildItem indexBuildItem,
+            BuildProducer<ApplicationClassPredicateBuildItem> classPredicate,
+            List<AdditionalSecuredClassesBuildIem> additionalSecuredClasses) {
+        classPredicate.produce(new ApplicationClassPredicateBuildItem(new SecurityCheckStorage.AppPredicate()));
+
+        Set<ClassInfo> additionalSecured = new HashSet<>();
+        if (additionalSecuredClasses != null) {
```

Comment:
This will never be null

Reply:
I'll be changing this in a subsequent PR anyway